components/lib/metergroup/MeterGroup.js

```diff
@@ -0,0 +1,159 @@
+import { useContext } from 'react';
+import { useMergeProps } from '../hooks/Hooks';
+import { PrimeReactContext } from '../api/Api';
+import { DomHandler, ObjectUtils, classNames } from '../utils/Utils';
+import { MeterGroupBase } from './MeterGroupBase';
+import { useHandleStyle } from '../componentbase/ComponentBase';
+
+export const MeterGroup = (inProps) => {
+    const context = useContext(PrimeReactContext);
+    const props = MeterGroupBase.getProps(inProps, context);
+    const { values, min, max, orientation, labelPosition, start, end, meterRenderer, labelListRenderer } = props;
+
+    const mergeProps = useMergeProps();
+    const { ptm, cx, isUnstyled } = MeterGroupBase.setMetaData({
+        props,
+        ...props.__parentMetadata,
+        context: {
+            disabled: props.disabled
+        }
+    });
+
+    useHandleStyle(MeterGroupBase.css.styles, isUnstyled, { name: 'progressbar' });
+
+    let totalPercent = 0;
+    let precentages = [];
+
+    values.map((item) => {
+        totalPercent += item.value;
+        precentages.push(Math.round((item.value / totalPercent) * 100));
+    });
+
+    const calculatePercentage = (meter = 0) => {
+        const percentageOfItem = ((meter - min) / (max - min)) * 100;
+
+        return Math.round(Math.max(0, Math.min(100, percentageOfItem)));
+    };
+
+    const rootProps = mergeProps(
+        {
+            className: classNames(props.className, cx('root', { orientation }))
+        },
+        MeterGroupBase.getOtherProps(props),
+        ptm('root')
+    );
+
+    const createMeters = () => {
+        const meters = values.map((item, index) => {
+            const meterInlineStyles = {
+                backgroundColor: item.color,
+                width: orientation === 'horizontal' ? calculatePercentage(item.value) + '%' : 'auto',
```

Comment:
Move `calculatePercentage(item.value)` into a variable and reuse that variable 3 times in this 1 method.